### DISCUSSION TITLE
Copilot failing to load with Error message

### DISCUSSION THREAD

**Discussion by lede701** [community/community#114615] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Visual Studio Pro  v17.9.199.22661
Windows 10 OS Build 19045.4239
Extension:
GitHub Copilot v1.156.0.0
GitHub Copilot Chat v0.2.755.36570

When opening older solutions I get the following error message any time I try to use Copilot chat:
![image](https://github.com/community/community/assets/7662065/b548db39-0b38-46dc-a96d-1933137d9a67)

**Log Error Message for Copilot Chat**
[Conversations Information] Copilot chat version 0.2.755-beta+8eda486b33.RR (0.2.755.36570). VS: VisualStudio.17.Release/17.9.5+34723.18. Session: b0608052-9747-4d7c-a9df-d8e7cadf176c. 
[Conversations Information] Need to get an auth token.
[ConversationsServiceManager Information] Out-of-process service finished initializing.
[Conversations Error] Skipping initializing ExtensionPart based agent providers due to incompatible Visual Studio version.
[Conversations Information] Status Code: 200, Reason Phrase: OK
[Conversations Information] Response Content: HasToken: True, ChatEnabled: True, ExpiresAt: 1711640187, RefreshIn: 1500, SKU: trial_30_monthly_subscriber, CopilotExclusion: False
[Conversations Information] Obtained a token which is valid for 00:24:59.9980013.
[PersistedCopilotSessionRepository Information] Persisted sessions are stored under E:\...\copilot-chat\b845f4b2\sessions
[CopilotSessionProvider Information] Begin sending message (ConversationId:747ef1aa-572c-4ac3-9e50-3345fbe22aaa, CorrelationId:b1c1a4cd-6dae-4bd2-8bed-306598a77ade, MessageId: 27b4d162-aa0a-4a25-8e3a-6edd39e95418)
[CopilotSessionProvider Information] Request content: "Hello are you out there?"
[CopilotSessionProvider Error] Error on conversation request (CorrelationId: b1c1a4cd-6dae-4bd2-8bed-306598a77ade): Microsoft.ServiceHub.Framework.ServiceActivationFailedException: Activating the "Microsoft.VisualStudio.Copilot.EnvironmentScope (0.3)" service failed. ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.VisualStudio.VCProjectEngine, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   at Microsoft.VisualStudio.Copilot.EnvironmentScope.AddProject(Object project)
   at Microsoft.VisualStudio.Copilot.EnvironmentScope.<InitializeAsync>d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.ServiceBroker.ServiceBrokerOfExportedServices.<Microsoft-ServiceHub-Framework-IServiceBroker-GetProxyAsync>d__13`1.MoveNext()
   --- End of inner exception stack trace ---
   at Microsoft.VisualStudio.Shell.ServiceBroker.ServiceBrokerOfExportedServices.<Microsoft-ServiceHub-Framework-IServiceBroker-GetProxyAsync>d__13`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.VisualStudio.Utilities.ServiceBroker.GlobalBrokeredServiceContainer.View.<GetProxyAsync>d__19`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.VisualStudio.PlatformUI.Packages.ServiceBroker.ServiceBrokerMefExport.<GetProxyAsync>d__6`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Copilot.CopilotSession.<AddCppEnvContextAsync>d__39.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Copilot.CopilotSession.<SendRequestCoreAsync>d__38.MoveNext()

**Log message for Copilot**
GitHub Copilot 1.156.0.0 (v1.156.0.0@82fafb995)
GitHub Copilot Agent 1.156.0 (1.156.0)
[DEBUG] [agent] [2024-03-28T15:06:44.303Z] Agent service starting
[DEBUG] [agent] [2024-03-28T15:06:44.799Z] Telemetry initialized
Auth Status: OK
Successfully authorized GitHub Copilot.

This only seems to happen on projects that are using older dotnet libraries.  I've tested this on dotnet core 7 projects and dotnet core 8 which work fine but anything older it give the provided error message.

### Update
TBGBrock solution to install the C++ developer tools does help this problem but it does put a road block in place for those of us who are C# developers and adds an extra step to our dev environment.

**Comment by TBGBrock** [comment]
I had the same problem, starting just a couple hours ago. Logs were very similar, with the missing "Microsoft.VisualStudio.VCProjectEngine". I repaired VS, removed the .vs folder and packages folder, and rebuilt the affected solution. None of those steps affected the problem.

A little research indicated that VCProjectEngine is used by C++ projects, which mine is not, so I did not have anything for C++ installed. However, adding a workload to VS 2022 that included C++ in its name resolved the problem. Not an ideal solution, but it works for now.

**Reply by lede701** [reply]
This is a good short term fix.  I used to do a ton of C++ programming but now I rarely get a chance so not sure why I didn't think to try this solution.  Thanks for the tip!

Now I can get back to questioning my regular scheduled programming.

**Comment by ForareVictor** [comment]
I have no errors at all only: 
"[Conversations Information] Copilot chat version 0.2.755-beta+8eda486b33.RR (0.2.755.36570). VS: VisualStudio.17.Release/17.9.5+34723.18. Session: 7b306727-5f69-40df-8bca-a560284636f5. 
[ConversationsServiceManager Information] Out-of-process service finished initializing.
[Conversations Error] Skipping initializing ExtensionPart based agent providers due to incompatible Visual Studio version."

**Reply by lede701** [reply]
Do you have the C++ runtime library install or C++ developer tools?

**Reply by ForareVictor** [reply]
I tried installing some yes, anyone in particular you suggest?
I am installing some other ones right now.

**Reply by lede701** [reply]
I just install the C++ tools:

![image](https://github.com/community/community/assets/7662065/4b3954fb-6fac-47a6-b416-dec61f4cfc70)

Once this finished I was able to run GitHub Copilot chat in my older projects.  Granted I still was able to get it to crash on asking it to help optimize one of my long winded API calls but that is a different issue.

**Reply by ForareVictor** [reply]
just installed that one same message but it works now when I open the chat window it seems

**Reply by lede701** [reply]
Odd and in the GitHub Copilot Chat logs there is no error?  That is very odd because mine got very chatty when it wasn't working.  Have you tried looking into the log file?  I think it is in your users folder AppData\Local\Temp\VSGitHubCopilotLogs, look at any of these files may provide some helpful insight.  Without an error it maybe hard to diagnose the issue and you could be just spinning your wheel when it could be something else.

**Reply by LoupaLoupa** [reply]
Error message is not descriptive enough!
